### PR TITLE
Parse APIException messages

### PR DESCRIPTION
**Pull request by smarlowucf**
Prevent json data from being embedded in csp-config error messages. This makes messages more digestible downstream.

### CODE REVIEW

File: csp_billing_adapter_k8s/plugin.py [hunk 1]
```diff
@@ -92,7 +108,7 @@ def save_cache(config: Config, cache: dict):
             return None  # Already exists
         else:
             log.error(f'Failed to save cache: {str(error)}')
-            raise
+            _re_raise_api_exception(error)
```
Comment (rtamalin) [inline csp_billing_adapter_k8s/plugin.py:111]: So technically this is raising a new exception, with a potentially new error message, whereas previously this was re-raising the original exception, which would come with the original traceback info...
Raising a new exception here means that we lose that original traceback data, and start with a traceback from the point where we raise the new exception...
Are we happy to lose the additional traceback data, from a debugging/support perspective?
Or would it be better to raise the new exception "from" the old exception, which, IIRC, would retain the original exception's traceback?

File: tests/unit/test_plugin.py [hunk 2]
```diff
@@ -50,6 +51,15 @@
 }
 
 
+def get_exception(status: int):
```
Comment (rtamalin) [inline tests/unit/test_plugin.py:54]: Minor nit, can be ignored.

Perhaps this should be `create_exception()`, or `gen_exception()`; "get" comes with a connotation of lookup/retrieval, whereas this is creating an exception object...

File: csp_billing_adapter_k8s/plugin.py (hunk 1 above)
Comment (smarlowucf) [reply csp_billing_adapter_k8s/plugin.py:111]: Not sure if it is of any help but raised from the error anyway.

**Review by rjschwei — APPROVED** [review]

**Review by KeithMnemonic — APPROVED** [review]
LGTM, did you want to squash this into one commit?

**Pull request merged**